quintilesims/layer0
# Upload tfstate to s3 by default when 'l0-setup apply' is executed

**What does this pull request do?**
v0.10.0 no longer uploaded the tfstate generated locally to s3. This PR changes that behavior to push the tfstate to s3 by default. Additionally an optional flag `--push=false` has been enabled which can be used to maintain existing behavior when executing `l0-setup apply <name>`.

**How should this be tested?**
Create a new Layer0 instance by executing the following commands:
```
l0-setup init test
l0-setup apply --push=false test
```

Confirm the s3 bucket created for the new Layer0 instance doesn't contain tfstate files.

Execute the apply command again, without the `--push` flag this time.
```
l0-setup apply test
```

Confirm the s3 bucket created for the previously created Layer0 instance does contain tfstate files.

**Checklist**


closes #271
